packages/core/src/actions/writeContract.ts

```diff
@@ -101,8 +102,11 @@ export async function writeContract<
   else
     client = await getConnectorClient(config, { account, chainId, connector })
 
+  const { connector: activeConnector } = getAccount(config)
+
   let request
-  if (__mode === 'prepared') request = rest
+  if (__mode === 'prepared' || activeConnector?.supportsSimulation)
+    request = rest
   else {
     const { request: simulateRequest } = await simulateContract(config, {
```

Comment:
This looks wrong

Reply:
?

Reply:
If a connector supports simulation, then this skips it

Reply:
Correct

Reply:
To be clear, we want to skip simulation if the Wallet supports it.

- If Wallet **does** support simulation, then Wagmi **skips the simulation** since the Wallet perform simulation instead.
- If Wallet **does not** support simulation, then Wagmi **performs the simulation**.